### ISSUE TITLE
Move new CSRF docs from Classic UI to Backend?

### ISSUE THREAD

**Issue by stevepiercy** [plone/volto#4232]
@plone/volto-team did not respond in a timely manner to https://github.com/plone/documentation/pull/1414, so I'm opening this issue here to ask the questions:

We created a new page [Cross-Site Request Forgery (CSRF)](https://6.docs.plone.org/classic-ui/csrf.html) which is located under "Classic UI".

1. Should this be moved under "Backend"?
2. Should it be updated with any content specific to Volto?

**Comment by tiberiuichim** [comment]
Sorry, I didn't see that there was input required on the other ticket.

There's no CSRF protection, as far as I know. @sneridagh maybe you know more about this?

**Comment by sneridagh** [comment]
There is nothing related in the Volto side. There is a good explanation on why this is not needed in the frontend, I can't elaborate now a complete one, but we can try to fill in the blanks. I think the gist of it is that you can't tamper in a link the headers needed to be injected in the API call (token, Accept). Neither the CORS allows you to send anything under another domain thet is JS-based.

@stevepiercy so go ahead and move it under backend.

**Comment by stevepiercy** [comment]
OK, I'll move it under backend. Let's keep this issue open, though, to ensure we do fill in the blanks.

Here's an abridged curl request when logging into https://demo.plone.org/ with potentially relevant headers that might help fill in the blanks for Volto.

```curl
curl 'https://demo.plone.org/++api++/@login' \
  -H 'authority: demo.plone.org' \
  -H 'accept: application/json' \
  -H 'origin: https://demo.plone.org' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-gpc: 1' \
```

We also need to change some language in the CSRF page, when it gets moved under Backend. It talks about [Plone doing automatic CSRF protection](https://6.docs.plone.org/classic-ui/csrf.html#auto-protection), but it sounds like that is done only in Classic UI, not "Plone". Plone 6 has two frontends, so we need to explicitly state it applies only to Classic UI. Is that correct? @jensens?

**Comment by jensens** [comment]
In the first place Plone does the hard-protection in the backend on database level, no matter who writes (ClassicUI, Volto, Scripts, name it).  It is a subscriber on the before-commit event and it gets grip unless it is disabled like described in the docs. 

So, question is 
- is it disabled for RestAPI somewhere? 
- If not, how is it managed?

And whatever is the answer, it needs a good piece of documentation.

**Comment by tiberiuichim** [comment]
plone.restapi has quite a few lines like this: https://github.com/plone/plone.restapi/blob/b33eeb990fb33957d4a85827ee23cb3aa07368cc/src/plone/restapi/services/content/add.py#L44-L45

**Comment by stevepiercy** [comment]
> It is a subscriber on the before-commit event and it gets grip unless it is disabled like described in the docs.

I'm sorry, "gets grip"? Do you mean "takes hold" or "takes effect"?

> plone.restapi has quite a few lines like this: https://github.com/plone/plone.restapi/blob/b33eeb990fb33957d4a85827ee23cb3aa07368cc/src/plone/restapi/services/content/add.py#L44-L45

That looks like CSRF _may_ be disabled, not _is absolutely_ disabled.

https://github.com/plone/plone.protect/blob/master/plone/protect/interfaces.py#L22

**Comment by tiberiuichim** [comment]
@stevepiercy as far as I understand it, CSRF is specifically disabled by plone.restapi in most of its code that commits to database: add/update content, etc. @tisto can you confirm this?

**Comment by davisagli** [comment]
My understanding is that CSRF checks are not really needed for a JSON web service _as long as_ the service is adequately validating the request's Content-Type (to make sure the attacker isn't sending JSON from a web form misidentified as text/plain) and as long as the server hasn't opened up the CORS policy for XHR requests too widely.

**Comment by davisagli** [comment]
(I'm curious why plone.restapi is disabling CSRF in specific endpoints rather than doing it generally in plone.rest though. It seems like something that should be handled at a framework level.)

**Comment by sneridagh** [comment]
@davisagli I vaguely remember that there was a reason for that but I can’t remember now. Maybe @tisto can recall it.

**Comment by jensens** [comment]
> I'm sorry, "gets grip"? Do you mean "takes hold" or "takes effect"?

@stevepiercy Sorry, should have looked this up, yes. 

According to what @davisagli says I tend to think we are doing it wrong. I think we better should not disable protection in restapi at all, but add a check in plone.protect if it is a ReST request and CORS is configured properly and onmy then allow writes without the header set.

**Comment by nitish-singh07** [comment]
is this issue still open ?@stevepiercy

**Comment by stevepiercy** [comment]
@nitish-singh07 yes it is still open, by virtue of the green icon with "Open" in its name. See Item 4 under [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do).

However the maintainers have not reached an agreement on what, if anything, needs to be written for documentation.

Because of ongoing discussion, this is not a good item for first-time contributors, so I changed its difficulty level to `42 lvl: moderate`.

**Comment by Manik-Khajuria-5** [comment]
hey @stevepiercy  Should I also open a PR in plone/documentation for this ?

**Comment by stevepiercy** [comment]
@Manik-Khajuria-5 no, see https://6.docs.plone.org/contributing/documentation/index.html#editing-external-package-documentation.

**Comment by Manik-Khajuria-5** [comment]
Okay, I’ll look into this.

**Comment by Manik-Khajuria-5** [comment]
@stevepiercy any updates on this PR Is there something I need to do ?

**Comment by stevepiercy** [comment]
@Manik-Khajuria-5 you could stop asking for updates and stop pinging me and other individuals. I'd appreciate it. Volto code is a very low priority for me. I care about documentation, translations, and accessibility.

Pull requests will get reviewed when the people who care about the issue have the time available and according to their priority.

**Comment by davisagli** [comment]
This is not ready for documentation work. There's a lot of speculation from experts above and some suggestions about how to improve the protection in Volto, but no definitive answers yet.

**Comment by Manik-Khajuria-5** [comment]
Understood on the priorities and the review process  I’ll avoid further pings and wait for reviews .I also understand that this isn’t ready for documentation .
